test/fork/migration-v2.js

```diff
@@ -655,55 +667,100 @@ describe('V2 upgrade', function () {
+    // Withdraw CLA stakes for all members
+    for (const member of Object.keys(CLA_STAKES_OUTPUT)) {
+      const memberBalanceBefore = await this.nxm.balanceOf(member);
+      await this.tokenController.withdrawClaimAssessmentTokens([member]);
+      const memberBalanceAfter = await this.nxm.balanceOf(member);
+      expect(memberBalanceAfter.sub(memberBalanceBefore)).to.be.equal(CLA_STAKES_OUTPUT[member]);
+    }
+
+    const tcBalanceAfter = await this.nxm.balanceOf(this.tokenController.address);
+    const tcBalanceDiff = tcBalanceBefore.sub(tcBalanceAfter);
+    expect(tcBalanceDiff).to.be.equal(this.claStakesSum);
 
     console.log({
-      governanceRewardsMigrated: governanceRewardsMigrated.toString(),
-      governanceRewardsSum: this.governanceRewardsSum.toString(),
+      tcBalanceBefore: formatEther(tcBalanceBefore),
+      tcBalanceAfter: formatEther(tcBalanceAfter),
+      tcBalanceDiff: formatEther(tcBalanceDiff),
+      claStakesSum: formatEther(this.claStakesSum),
+      tcBalanceDiffMinusCLAStakes: formatEther(tcBalanceDiff.sub(this.claStakesSum)),
     });
 
-    /*
-      -1106654884061072517264
-      +870391213513961173071
-
-      Extra tokens:
+    // Attempt to withdraw CLA stakes for all members again, and assert that their balances don't change
+    for (const member of Object.keys(CLA_STAKES_OUTPUT)) {
+      const memberBalanceBefore = await this.nxm.balanceOf(member);
+      await this.tokenController.withdrawClaimAssessmentTokens([member]);
+      const memberBalanceAfter = await this.nxm.balanceOf(member);
+      expect(memberBalanceAfter.sub(memberBalanceBefore)).to.be.equal(0);
+    }
+  });
 
-      236.26367054711136 NXM
-     */
-    // expect(governanceRewardsMigrated).to.be.equal(this.governanceRewardsSum);
+  it('Get CN locked amount', async function () {
+    await getCNLockedAmount(ethers.provider);
   });
 
-  // TODO review
-  it.skip('check if TokenController balance checks out with Governance rewards', async function () {
-    const tcNxmBalance = await this.nxm.balanceOf(this.tokenController.address);
+  it('Check all members with CN locked NXM can withdraw & TC has the correct balance afterwards', async function () {
+    const CN_LOCKED_AMOUNT_OUTPUT = require(CN_LOCKED_AMOUNT_OUTPUT_PATH);
+    const tcBalanceBefore = await this.nxm.balanceOf(this.tokenController.address);
+
+    // Withdraw CN token for all members
+    for (const lock of CN_LOCKED_AMOUNT_OUTPUT) {
+      const memberBalanceBefore = await this.nxm.balanceOf(lock.member);
+      await this.tokenController.withdrawCoverNote(lock.member, lock.coverIds, lock.lockReasonIndexes);
+      const memberBalanceAfter = await this.nxm.balanceOf(lock.member);
+      expect(memberBalanceAfter.sub(memberBalanceBefore)).to.be.equal(lock.amount);
 
-    const rewardsSum = this.governanceRewardsSum;
+      const lockReasonsCount = (await this.tokenController.getLockReasons(lock.member)).length;
+      expect(lockReasonsCount).lte(1);
```

Comment:
what is thee reason for asserting this ?

lockReasonCount <= 1

Reply:
When we withdraw CLA stakes, we don't delete the reasons, just mark the entry as "claimed"